src/plugins/memory/memory-tile.js

```diff
@@ -0,0 +1,92 @@
+import PropTypes from 'prop-types';
+import reactPlayerNs from 'react-player';
+import Markdown from '../../components/markdown.js';
+import { RESOURCE_TYPE } from '../../domain/constants.js';
+import ClientConfig from '../../bootstrap/client-config.js';
+import { analyzeMediaUrl } from '../../utils/media-utils.js';
+import { getAccessibleUrl } from '../../utils/source-utils.js';
+import { useService } from '../../components/container-context.js';
+import AudioIcon from '../../components/icons/general/audio-icon.js';
+import React, { Fragment, useEffect, useRef, useState } from 'react';
+
+let timeoutToPlayMedia;
+const ReactPlayer = reactPlayerNs.default || reactPlayerNs;
+
+function MemoryTile({ text, sourceUrl, playMedia }) {
+  const playerRef = useRef();
+  const isMounted = useRef(false);
+  const clientConfig = useService(ClientConfig);
+
+  const accessibleUrl = getAccessibleUrl({ url: sourceUrl, cdnRootUrl: clientConfig.cdnRootUrl });
+  const { resourceType } = sourceUrl ? analyzeMediaUrl(sourceUrl) : { resourceType: RESOURCE_TYPE.none };
+
+  const [isPlaying, setIsPlaying] = useState(playMedia);
+
+  useEffect(() => {
+    isMounted.current = true;
+    return () => {
+      isMounted.current = false;
+    };
+  }, []);
+
+  useEffect(() => {
+    if (!playMedia) {
+      setIsPlaying(false);
+      if (timeoutToPlayMedia) {
+        clearTimeout(timeoutToPlayMedia);
+      }
+      return;
+    }
+
+    playerRef.current?.seekTo(0);
+    timeoutToPlayMedia = setTimeout(() => {
+      if (isMounted.current && playMedia) {
+        setIsPlaying(true);
+      }
+    }, 500);
+
+  }, [playerRef, playMedia]);
```

Comment:
I was trying to figure out what this `useEffect` is doing in contrast to just using the prop `playMedia`. It also uses the the global variable `timeoutToPlayMedia` shared between all possible memory tiles on the page. Is this intentional?

Reply:
Though technically only the last flipped card should play and the previous should be stopped, actually this component should not directly impact the others. So yes, not a good place to put that variable, I've moved it into a `ref`.

Reply:
I was just thinking we might not need the whole `useEffect` block. I tried removing it and handing down the `playMedia` prop directly to the react player and it worked in my use case. But I haven't tested all cases, so I was wondering why we need it?

Reply:
We need the `useEffect` so that when you flip a card, the media starts playing with a small delay. Otherwise it starts playing immediately, before you even get to see the card completely face-up. I find that not to be a great experience, possibly even startling for the user.

Reply:
Ah, ok! Got it, thanks for the explanation. In this case I think it would be clearer to set `isPlaying` initially to `false` in any case in line 23, so that it is only set to `true` in a delayed manner within the `useEffect`. What do you think?

Reply:
Currently the media will play directly if the initial `playState` is `true` as there is no transition of the flip card to wait for. To me the delay makes sense in the case of actual state change, in this context. We actually only have the case of card flip, never of direct play, so I don't feel strongly in any direction. But I feel the intention of the code might be clearer this way? 🤷‍♀️

Reply:
The way it is now just has this strange sequence of events when you follow the case of `playMedia` set to `true` initially: `isPlaying` is set to `true` which makes the player play immediately. On the next tick (when `useEffect` is called the first time), we seek back to position 0 (`playerRef.current?.seekTo(0);`) and then 500 ms later we set it `isPlaying` to `true` again (which it already is). That made the code hard to understand for me. So even if I agree on the flip transition argument I would still set `isPlaying` to false initially to keep the code logic easier to follow. In our case it doesn't apply anyways.

Reply:
Ok, if this were ever the use case, the reset to timecode 0 would happen weirdly again mid play, that's true. I've set it to `false` then. Thanks